app/models/assignment.rb

```diff
@@ -622,6 +623,41 @@ def pair_programming_enabled?
     self.enable_pair_programming
   end
 
+  # Get next date when the assignment is due
+  def get_next_due_date(assignment_id, topic_id = nil)
+    assignment = Assignment.find(assignment_id)
+    if assignment.staggered_deadline?
+      next_due_date = TopicDueDate.find_by(['parent_id = ? and due_at >= ?', topic_id, Time.zone.now])
+      upcoming_due_dates_after_topic_date = assignment.upcoming_due_dates_after_topic_date(topic_id)
+      # if certion TopicDueDate is not exist, we should query next corresponding AssignmentDueDate.
+      # eg. Time.now is 08/28/2016
+      # One topic uses following deadlines:
+      # TopicDueDate      08/01/2016
+      # TopicDueDate      08/02/2016
+      # TopicDueDate      08/03/2016
+      # AssignmentDueDate 09/04/2016
+      # In this case, we cannot find due_at later than Time.now in TopicDueDate.
+      # So we should find next corresponding AssignmentDueDate, starting with the 4th one, not the 1st one!
+      if next_due_date.nil? && upcoming_due_dates_after_topic_date
+        next_due_date = upcoming_due_dates_after_topic_date.detect { |assignment_due_date| assignment_due_date.due_at >= Time.zone.now }
+      end
+    else
+      next_due_date = AssignmentDueDate.find_by(['parent_id = ? && due_at >= ?', assignment_id, Time.zone.now])
+    end
+    next_due_date
+  end
+
+  # Create separate method to get next dates after topic dates to follow single responsibility principle
+  def upcoming_due_dates_after_topic_date(topic_id = nil)
+    topic_due_date_size = TopicDueDate.where(parent_id: topic_id).size
+    upcoming_due_dates_after_topic_date = AssignmentDueDate.where(parent_id: id)[topic_due_date_size..-1]
+  end
+
+  # Get current date when the assignment is due
```

Comment:
Comment should read, "Return the next (upcoming) due date for the current assignment."